src/Microsoft.ApiDesignGuidelines.Analyzers/Core/EnumStorageShouldBeInt32.cs

```diff
@@ -25,25 +30,56 @@ public abstract class EnumStorageShouldBeInt32Analyzer : DiagnosticAnalyzer
                                                                              s_localizableMessageNotInt32,
                                                                              DiagnosticCategory.Design,
                                                                              DiagnosticSeverity.Warning,
-                                                                             isEnabledByDefault: false,
+                                                                             isEnabledByDefault: true,
                                                                              description: s_localizableDescription,
-                                                                             helpLinkUri: null,     // TODO: add MSDN url
-                                                                             customTags: WellKnownDiagnosticTags.Telemetry);
-        internal static DiagnosticDescriptor NotIntegralRule = new DiagnosticDescriptor(RuleId,
-                                                                             s_localizableTitle,
-                                                                             s_localizableMessageNotIntegral,
-                                                                             DiagnosticCategory.Design,
-                                                                             DiagnosticSeverity.Warning,
-                                                                             isEnabledByDefault: false,
-                                                                             description: s_localizableDescription,
-                                                                             helpLinkUri: null,     // TODO: add MSDN url
+                                                                             helpLinkUri: "https://msdn.microsoft.com/en-us/library/ms182147.aspx",
                                                                              customTags: WellKnownDiagnosticTags.Telemetry);
 
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(NotInt32Rule, NotIntegralRule);
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(NotInt32Rule);
 
         public override void Initialize(AnalysisContext analysisContext)
         {
-            
+            analysisContext.RegisterCompilationStartAction(compilationContext =>
+            {
+                var flagsAttribute = WellKnownTypes.FlagsAttribute(compilationContext.Compilation);
+                compilationContext.RegisterSymbolAction(symbolContext => AnalyzeSymbol(symbolContext, flagsAttribute), SymbolKind.NamedType);
+            });
+        }
+
+        private void AnalyzeSymbol(SymbolAnalysisContext context, INamedTypeSymbol flagsAttribute)
+        {
+            var symbol = (INamedTypeSymbol)context.Symbol;
+
+            // If not enum type exit
```

Comment:
A nit-pick, but just one of my coding preferences: If the comment doesn't do anything except repeat what the code does, I would omit it. I add comments when it's necessary to describe _why_ the code does what it does, or when the implementation is unclear (but then I always ask if it could be clearer). In this case, I'd omit it.